### PR TITLE
Initial dev release

### PR DESCRIPTION
**Pull request by ymmit85**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# VMware vROPs As Built Report Changelog
+
+## 0.1.0
+### What's New
+
+- Refactored code into new module as Tim Carman is really OCD about it.
```
Comment (tpcarman) [inline CHANGELOG.md]: Let's be serious now

File: README.md [hunk 2]
```diff
@@ -1,2 +1,126 @@
...
+- [VMware PowerCLI Module](https://www.powershellgallery.com/packages/VMware.PowerCLI/)
+- [PowervROPs Module](https://github.com/ymmit85/PowervROps/)
+
+### Module Installation
+
+Open a Windows PowerShell terminal window and install each of the required modules as follows;
+```powershell
+install-module AsBuiltReport
+install-module VMware.PowerCLI
+import-module PowervROPs
```
Comment (tpcarman) [inline README.md]: Additional info/steps required on how to install PowervROps

File: AsBuiltReport.VMware.vROPs.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        "Version": "1.0",
+        "Status": "Released"
+    },
+    "Options": {
+        "AuthSource": "local",
+        "AlertFilter": "*"
+    },
+    "HealthCheck": {},
+    "InfoLevel": {
+        "_comment_": "0 = Disabled, 1 = Summary, 2 = Informative, 3 = Detailed, 4 = Adv Detailed, 5 = Comprehensive",
```
Comment (tpcarman) [inline AsBuiltReport.VMware.vROPs.json]: Do you intend on having 5 InfoLevels? Only use the number of InfoLevels you need or intend on building.

File: README.md [hunk 4]
```diff
@@ -1,2 +1,126 @@
...
+| InfoLevel | Users | 3
+| InfoLevel | Adapters | 3
+| InfoLevel | RemoteCollectors | 3
+| InfoLevel | Alerts | 3
+| InfoLevel | Symptoms | 3
+| InfoLevel | SuperMetrics | 3
+| InfoLevel | ServiceStatus | 3
+| InfoLevel | CustomGroups | 3
+
+There are 6 levels (0-5) of detail granularity for each section as follows;
```
Comment (tpcarman) [inline README.md:84]: Update as per previous comment and your intentions for the vROps report

File: README.md [hunk 5]
```diff
@@ -1,2 +1,126 @@
...
+## Examples 
+- Generate HTML & Word reports with Timestamp
+Generate a vROps As Built Report for vROps Server 'vrops-01.corp.local' using specified credentials. Export report to HTML & DOC formats. Use default report style. Append timestamp to report filename. Save reports to 'C:\Users\Tim\Documents'
+```powershell
+New-AsBuiltReport -Target 'vrops-01.corp.local' -Username 'admin' -Password 'VMware1!' -Report VMware.vROps -Format Html,Word -OutputPath 'C:\Users\Tim\Documents' -Timestamp
+```
+- Generate HTML & Text reports with Health Checks
+Generate a vROps As Built Report for vROps Server 'vrops-01.corp.local' using stored credentials. Export report to HTML & Text formats. Use default report style. Highlight environment issues within the report. Save reports to 'C:\Users\Tim\Documents'
+```powershell
+New-AsBuiltReport -Target 'vrops-01.corp.local' -Credentials $Creds -Report VMware.vROps -Format Html,Text -OutputPath 'C:\Users\Tim\Documents' -EnableHealthCheck
```
Comment (tpcarman) [inline README.md]: Move `Report` parameter position to be first and update parameter from `Credentials` to `Credential`

File: README.md [hunk 6]
```diff
@@ -1,2 +1,126 @@
...
+\*\* *future release*
+
+### Healthcheck
+The **Healthcheck** sub-schema is used to toggle health checks on or off. Currently none configured.
+
+## Examples 
+- Generate HTML & Word reports with Timestamp
+Generate a vROps As Built Report for vROps Server 'vrops-01.corp.local' using specified credentials. Export report to HTML & DOC formats. Use default report style. Append timestamp to report filename. Save reports to 'C:\Users\Tim\Documents'
+```powershell
+New-AsBuiltReport -Target 'vrops-01.corp.local' -Username 'admin' -Password 'VMware1!' -Report VMware.vROps -Format Html,Word -OutputPath 'C:\Users\Tim\Documents' -Timestamp
```
Comment (tpcarman) [inline README.md]: Move `Report` parameter position to be first

File: README.md [hunk 7]
```diff
@@ -1,2 +1,126 @@
...
+```
+- Generate HTML & Text reports with Health Checks
+Generate a vROps As Built Report for vROps Server 'vrops-01.corp.local' using stored credentials. Export report to HTML & Text formats. Use default report style. Highlight environment issues within the report. Save reports to 'C:\Users\Tim\Documents'
+```powershell
+New-AsBuiltReport -Target 'vrops-01.corp.local' -Credentials $Creds -Report VMware.vROps -Format Html,Text -OutputPath 'C:\Users\Tim\Documents' -EnableHealthCheck
+```
+- Generate report with multiple vROps Servers using Custom Style
+Generate a single vROps As Built Report for vROps Servers 'vrops-01.corp.local' and 'vrops-02.corp.local' using specified credentials. Report exports to WORD format by default. Apply custom style to the report. Reports are saved to the user profile folder by default.
+```powershell
+New-AsBuiltReport -Target 'vrops-01.corp.local','vcenter-02.corp.local' -Username 'admin' -Password 'VMware1!' -Report VMware.vROps -StylePath C:\Scripts\Styles\MyCustomStyle.ps1
```
Comment (tpcarman) [inline README.md]: Move `Report` parameter position to be first

File: README.md [hunk 8]
```diff
@@ -1,2 +1,126 @@
...
+```
+- Generate HTML & Word reports, attach and send reports via e-mail
+Generate a vROps As Built Report for vROps Server 'vrops-01.corp.local' using specified credentials. Export report to HTML & DOC formats. Use default report style. Reports are saved to the user profile folder by default. Attach and send reports via e-mail.
+```powershell
+New-AsBuiltReport -Target 'vrops-01.corp.local' -Username 'admin' -Password 'VMware1!' -Report VMware.vROps -Format Html,Word -OutputPath C:\Users\Tim\Documents -SendEmail
```
Comment (tpcarman) [inline README.md]: Move `Report` parameter position to be first

Comment (ymmit85) [reply README.md]: Fixed

File: README.md (hunk 7 above)
Comment (ymmit85) [reply README.md]: fixed

File: README.md (hunk 6 above)
Comment (ymmit85) [reply README.md]: fixed

File: README.md (hunk 5 above)
Comment (ymmit85) [reply README.md]: fixed

File: AsBuiltReport.VMware.vROPs.json (hunk 3 above)
Comment (ymmit85) [reply AsBuiltReport.VMware.vROPs.json]: Have set it to max of 3.

File: README.md (hunk 4 above)
Comment (ymmit85) [reply README.md:84]: fixed

File: CHANGELOG.md (hunk 1 above)
Comment (ymmit85) [reply CHANGELOG.md]: Ok

File: README.md (hunk 2 above)
Comment (ymmit85) [reply README.md]: Fixed

**Pull request closed without merging**